src/py/flwr/cli/flower_toml_test.py

```diff
@@ -0,0 +1,308 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Test for Flower command line interface `run` command."""
+
+import os
+import textwrap
+from typing import Any, Dict
+
+from .flower_toml import (
+    load_flower_toml,
+    validate_flower_toml,
+    validate_flower_toml_fields,
+    validate_object_reference,
+)
+
+
+def test_load_flower_toml_load_from_cwd(tmp_path: str) -> None:
+    """Test if load_template returns a string."""
+    # Prepare
+    flower_toml_content = """
+        [project]
+        name = "fedgpt"
+
+        [flower.components]
+        serverapp = "fedgpt.server:app"
+        clientapp = "fedgpt.client:app"
+
+        [flower.engine]
+        name = "simulation" # optional
+
+        [flower.engine.simulation.super-node]
```

Comment:
In case we decide to keep this in `flower.toml` (SuperNode has only two spellings: `SuperNode` in text, or `supernode` in code):
```suggestion
        [flower.engine.simulation.supernode]
```